src/server/local-auth-helpers.js

```diff
@@ -0,0 +1,187 @@
+import AuthHasher from 'passport-local-authenticate'
+import { User, Invite, Organization } from './models'
+import { capitalizeWord } from './api/lib/utils'
+
+const errorMessages = {
+  invalidInvite: 'Invalid invite code. Contact your administrator.',
+  invalidCredentials: 'Invalid username or password',
+  emailTaken: 'That email is already taken.',
+  passwordsDontMatch: 'Passwords don\'t match.',
+  invalidResetHash: 'Invalid username or password reset link. Contact your administrator.',
+  noSamePassword: 'Old and new password can\'t be the same'
+}
+
+const validUuid = async (nextUrl, uuidMatch) => {
+  let foundUUID
+  if (nextUrl.includes('join')) {
+    foundUUID = await Organization.filter({ uuid: uuidMatch[0] })
+  } else if (nextUrl.includes('invite')) {
+    foundUUID = await Invite.filter({ hash: uuidMatch[0] })
+  }
+  if (foundUUID.length === 0) {
+    return [null, false, errorMessages.invalidInvite]
+  }
+}
+
+const login = async ({
+  password,
+  existingUser,
+  nextUrl,
+  uuidMatch
+}) => {
+  if (existingUser.length === 0) {
+    return [null, false, errorMessages.invalidCredentials]
+  }
+
+  // Verify UUID validity
+  if (nextUrl) validUuid(nextUrl, uuidMatch)
+
+  // Get salt and hash and verify user password
+  const pwFieldSplit = existingUser[0].auth0_id.split('|')
+  const hashed = {
+    salt: pwFieldSplit[1],
+    hash: pwFieldSplit[2]
+  }
+  return new Promise((resolve, reject) => {
+    AuthHasher.verify(
+      password, hashed,
+      (err, verified) => {
+        if (err) reject(err)
+        if (verified) {
+          resolve([null, existingUser[0]])
+        }
+        resolve([null, false, errorMessages.invalidCredentials])
+      }
+    )
+  })
+}
+
+const signup = async ({
+  lowerCaseEmail,
+  password,
+  existingUser,
+  nextUrl,
+  uuidMatch,
+  reqBody
+}) => {
+  // Verify UUID validity
+  if (nextUrl) validUuid(nextUrl, uuidMatch)
```

Comment:
Remember nextUrl is user input controlled -- I think for signup we *always* want to validate a Uuid, right?  so if there's no nextUrl, we can/should fail-fast -- but that might be better done in validUuid.

Also, we need an `await` here -- and also need to test whether it succeeded and maybe send back the error?  (based on the template below)